### PR TITLE
Add `LinkedGameObjectMap` API

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Added an API to expose the `EntityId -> [GameObject]` relationship. This is wrapped in an object to ensure that we expose a list which is readonly (we really don't want users to mutate the underlying data). 

Also added a subscription manager for this (only in the GameObjectCreation module as the linker actually lives in a system here!) and added tests for this. 

**Commits are ordered and logical!**

#### Tests
Added some tests for the subscription fulfillment. 

#### Documentation
- [x] Changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 ## Unreleased
 
+### Added
+
+- Added the `LinkedGameObjectMap` class for finding the `GameObject`(s) linked with a specified `EntityId`.
+    - This can be used with the `[Require]` annotation to inject it into your `MonoBehaviours` provided you are using the `GameObjectCreation` feature module.
```
Comment (zeroZshadow) [inline CHANGELOG.md]: I would rephrase this line a bit to make it clearer what you put require on to have this work.

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/LinkedGameObjectMap.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core;
+using UnityEngine;
+
+namespace Improbable.Gdk.Subscriptions
+{
+    /// <summary>
+    ///     Represents the mapping between SpatiaLOS entity IDs and linked GameObjects.
```
Comment (paulbalaji) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/LinkedGameObjectMap.cs]:
```suggestion
    ///     Represents the mapping between SpatialOS entity IDs and linked GameObjects.
```

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/LinkedGameObjectMap.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        ///     if any are linked; otherwise, null. This parameter is passed uninitialized.
+        /// </param>
+        /// <returns>True, if there are any GameObjects linked to the <see cref="EntityId"/>; otherwise false</returns>
+        public bool TryGetLinkedGameObjects(EntityId entityId, out IReadOnlyList<GameObject> linkedGameObjects)
+        {
+            linkedGameObjects = GetLinkedGameObjects(entityId);
+            return linkedGameObjects != null;
+        }
+    }
+
```
Comment (paulbalaji) [inline workers/unity/Packages/com.improbable.gdk.core/Subscriptions/LinkedGameObjectMap.cs]: nit: empty lines

File: workers/unity/Packages/com.improbable.gdk.core/Subscriptions/LinkedGameObjectMap.cs (hunk 2 above)
Comment (jamiebrynes7) [reply workers/unity/Packages/com.improbable.gdk.core/Subscriptions/LinkedGameObjectMap.cs]: Its actual spatia line-of-sight

File: CHANGELOG.md (hunk 1 above)
Comment (jamiebrynes7) [reply CHANGELOG.md]: Added an example at the end?

Comment (zeroZshadow) [reply CHANGELOG.md]: 👍
